python/pdo/service_client/enclave.py

```diff
@@ -70,26 +84,50 @@ def _attach_storage_service_(self, storage_service) :
         self.check_blocks = storage_service.check_blocks
 
     # -----------------------------------------------------------------
-    # encrypted_session_key -- base64 aes key encrypted with enclave's rsa key
-    # encrypted_request -- base64 string encrypted with aes session key
+    # encrypted_session_key -- byte string containing aes key encrypted with enclave's rsa key
+    # encrypted_request -- byte string request encrypted with aes session key
     # -----------------------------------------------------------------
-    def send_to_contract(self, encrypted_session_key, encrypted_request) :
-        request = dict()
-        request['encrypted_session_key'] = encrypted_session_key
-        request['encrypted_request'] = encrypted_request
-
+    def send_to_contract(self, encrypted_session_key, encrypted_request, encoding='base64') :
+        request_identifier = self.request_identifier
+        self.request_identifier += 1
         try :
             url = '{0}/invoke'.format(self.ServiceURL)
-            response = requests.post(url, json=request, timeout=self.default_timeout)
```

Comment:
why is the 429 response code not checked for update requests? Arent the update requests the most time consuming requests and hence the reason to keep a sever busy when there are more than 8 clients?

Reply:
429 is dropped completely with the WSGI version

Reply:
I guess I then need to understand how WSGI helps to manage more than concurrent 10 clients are handled without client side retry and no queue at server.

Reply:
There is a queue at the server. The difference is that the wait is on the network connection at the server rather than a wait at the client. Until we can expose the number of active threads on the server, there is no easy way to implement the 429 response.